EDGAR/parser.py

```diff
@@ -305,51 +306,56 @@ def get_page_number(self, page_location: dict, element: WebElement) -> int:
-            
+        df = pd.DataFrame(columns=COLUMN_NAMES)
+        text_on_page = self.parse_text_by_page()
+        page_location = self.find_page_location()
+
+        #print(len(page_location),len(text_on_page))
+        number_Null = 0
+        for i, elem in enumerate(webelements):
             default_dict = {attribute: np.nan for attribute in COLUMN_NAMES}
             page_num, y = self.get_page_number(page_location, elem)
 
-            default_dict.update({"found_index": int(idx),"full_text": elem.text, "is_annotated": False,
-                                "x": elem.location["x"], "y": y, "page_number": page_num, "name": elem.get_attribute('name'),
-                                "height": elem.size["height"], "width": elem.size["width"], "in_table": in_table[idx]})
+            if(page_num == None):
+                number_Null += 0
+                continue
+            default_dict.update({"anno_text": np.nan, "found_index": int(i),"span_text": elem.text,
+                                "page_text":text_on_page[page_num]['text'], "is_annotated": 0,
+                                "x": elem.location["x"], "y": y, "page_number": page_num,
+                                "height": elem.size["height"], "width": elem.size["width"], "in_table": int(in_table[i]}))
 
             count = 0
-
-            new_df = pd.DataFrame(columns=COLUMN_NAMES).astype({"in_table":bool,"is_annotated":bool}) 
             
+            new_df = pd.DataFrame(columns=COLUMN_NAMES)
             for j, annotation in enumerate(annotations[elem]):
                 new_dict = default_dict.copy()
-                
-                val = {"anno_index": j , "x": annotation.location["x"], "is_annotated": True,
-                        "label_text": annotation.text, "anno_ix_type": annotation.tag_name}
-                
+
+                val = {"anno_index": j , "x": annotation.location["x"], "is_annotated": 1,
+                        "anno_text": annotation.text, "anno_ix_type": annotation.tag_name}
+
                 val["page_number"], val["y"] = self.get_page_number(page_location, annotation)
 
                 for _attr in ["name", "id", "contextref", "decimals", "format", "unitref"]:
                     val[f"anno_{_attr}"] = annotation.get_attribute(_attr)
                 for _size in ["width", "height"]:
                     val[_size] = annotation.size[_size]
-                
+
                 new_dict.update(val)
-                temp_df = pd.DataFrame(new_dict, index=[0]).astype({"in_table":bool, "is_annotated":bool})
+                temp_df = pd.DataFrame(new_dict, index=[0])
                 new_df = pd.concat([temp_df, new_df], ignore_index=True)
-                
+
                 count += 1
-            
+
 
             if count == 0:
-                new_df = pd.DataFrame(default_dict, index=[0]).astype({"in_table":bool, "is_annotated":bool})
+                new_df = pd.DataFrame(default_dict, index=[0])
```

Comment:
Okay so I am seeing a decision here to treat booleans as integers -> is that common Pandas practice? Does pandas set datatype 'per column'? Is the distinction something that is important later?

Reply:
I did push a change to change int(in_table == True) to int(in_table) for readability reasons as in_table is strictly a boolean array and it's best practice to not do ==True on a boolean object

Reply:
I added to the bottom. otherwise, it generate 

> /Users/yitingli/Desktop/EDGAR/FinDocNLP/setup/../EDGAR/parser.py:359: FutureWarning: In a future version, object-dtype columns with all-bool values will not be included in reductions with bool_only=True. Explicitly cast to bool dtype instead.
  df = pd.concat([new_df,df], ignore_index=True)


<img width="698" alt="Screen Shot 2022-11-21 at 3 13 33 PM" src="https://user-images.githubusercontent.com/66972539/203149756-3240c468-b6c2-45ac-b818-5bdea20d6c49.png">

Reply:
Sorry to confirm: you added astype(bool) for the df columns?

I think that is a safe future-friendly way to handle it --> explicitly cast df.astype bool for the 'bool' columns.